tollway/vehicle.py

```diff
@@ -0,0 +1,102 @@
+from random import choice
+from collections import deque
+
+import requests
+from bs4 import BeautifulSoup, Tag
+from faker import Faker
+from faker_vehicle import VehicleProvider
+from us.states import STATES_AND_TERRITORIES
+
+
+STATE_NAMES = [state.name for state in STATES_AND_TERRITORIES]
+TOLLWAYS_URL = "https://en.wikipedia.org/wiki/List_of_toll_roads_in_the_United_States"
+
+
+def create_vehicle(fake: Faker) -> dict:
+    
+    vehicle = {f"{key.lower()}": value for key, value in fake.vehicle_object().items()}
+    vehicle["license_plate"] = fake.license_plate()
+    vehicle["vin"] = fake.vin()
+    vehicle["state"] = choice(STATES_AND_TERRITORIES).abbr
+    vehicle["primary_color"] = fake.color_name()
+
+    return vehicle
+
+
+def _num_of_tables_between_states(soup: BeautifulSoup) -> dict:
+    states = {}
+    table_count_between_states = 0
+    state_name = ""
+    for row in soup.contents[0]:
+        if not isinstance(row, Tag):
+            continue
+
+        if isinstance(row, Tag) and row.name == "h2" and row.span.string in STATE_NAMES:
+            state_name = row.span.string
+            table_count_between_states = 0
+
+        if isinstance(row, Tag) and row.name == "table" and row["class"][0] == "wikitable" and state_name != "":
+            table_count_between_states += 1
+            states[state_name] = table_count_between_states
+
+    return states
+
+
+def _state_tollways(soup: BeautifulSoup) -> list:
+    names = []
+
+    for table in soup.find_all("table", class_="wikitable"):
+        table_rows = table.tbody.find_all("tr")[1:]
+        state_tollways = []
+        for table_row in table_rows:
+            for tag in table_row.find_all("a"):
+                if tag.string is None:
+                    continue
+                state_tollways.append(tag.string)
+                break
+        names.append(state_tollways)
+    
+    return names
+
+
+def get_tollways() -> dict:
+    tollways_html = requests.get(TOLLWAYS_URL)
```

Comment:
you could pass the html into the function as arg to decouple the request call, makes it easier to test